tools/driver/CMakeLists.txt

```diff
@@ -157,15 +182,6 @@ swift_create_post_build_symlink(swift-frontend
   DESTINATION "swift-parse-test${CMAKE_EXECUTABLE_SUFFIX}"
   WORKING_DIRECTORY "${SWIFT_RUNTIME_OUTPUT_INTDIR}")
 
-add_swift_tool_symlink(swift swift-frontend compiler)
```

Comment:
As far as I can tell what happened here is that https://github.com/apple/swift/pull/6053 added this and removed the others, but that was somehow lost in the conflict resolution in https://github.com/apple/swift/pull/10337. I haven't looked at the LLVM closely, but I assume we should probably prefer `add_swift_tool_symlink` over `swift_create_post_build_symlink` and `swift_install_in_component`? Any thoughts @edymtt/@compnerd?

Reply:
One thing about `add_swift_tool_symlink` is that it requires the "source" be a component. So we can't simply call:
```
add_swift_tool_symlink(swift swift-driver compiler)
```
Because that leads to:
```
CMake Error at /Volumes/Data/GHWorkspace/build/Ninja-Release/llvm-macosx-arm64/lib/cmake/llvm/AddLLVM.cmake:2168 (get_target_property):
  get_target_property() called with non-existent target "swift-driver".
```

Reply:
@bnbarham yes, we do prefer the `add_swift_tool_symlink` because it correctly handles the _install_ of the link (or in the case of Windows, copy 😠).  This makes it significantly better for building the toolchain distribution.  Without that we will need to add a whole slew of custom install logic.

Reply:
To be clear, we have that logic already (we have both today):
```
swift_install_in_component(FILES "${SWIFT_RUNTIME_OUTPUT_INTDIR}/swift${CMAKE_EXECUTABLE_SUFFIX}"
                           DESTINATION "bin"
                           COMPONENT compiler)
```

> One thing about add_swift_tool_symlink is that it requires the "source" be a component. So we can't simply call:

We could fix that by adding a custom target for the early swift driver (which does the copy) right? ie. change `swift_create_early_driver_copies` to instead use `add_custom_command` + `add_custom_target` rather than `configure_file`.

Reply:
We already have a slew of custom install logic though, with `swift_install_in_component` later in this file. 

We can pursue a refactor here to replace both `swift_create_post_build_symlink` and `swift_install_in_component` with LLVM tooling, but as-is, the logic is already there, and `add_swift_tool_symlink` would not work for this use-case for the reason described above.

Reply:
Also, `swift_install_in_component` does, I believe, already do Windows-specific things (`copy` instead of `symlink`) so I don't think this will functionally change much here.